src/Data/Swagger/Internal/AesonUtils.hs

```diff
@@ -344,3 +349,55 @@ sopSwaggerGenericToEncoding'' (SwaggerAesonOptions prefix _ sub) = go
       where (x, y) = span isUpper s
 
 #endif
+
+percentEncodeS :: String -> String
```

Comment:
I would use [urlEncode](https://hoogle.zinfra.io/file/root/.stack/snapshots/x86_64-linux/01af7b6eb1c680941d3ef08786685b31d2470c8b36e981c7bdad15d91f453def/8.6.5/doc/http-types-0.12.3/Network-HTTP-Types.html#v:urlEncode).

Reply:
Looking through the source code, it doesn't escape all reserved characters: [source](https://hoogle.zinfra.io/file/root/.stack/snapshots/x86_64-linux/01af7b6eb1c680941d3ef08786685b31d2470c8b36e981c7bdad15d91f453def/8.6.5/doc/http-types-0.12.3/src/Network.HTTP.Types.URI.html#unreservedQS).

It doesn't escape the `sub-delims` set of characters from [RFC 3986](https://tools.ietf.org/html/rfc3986#section-2.2), which isn't strictly necessary and is scheme-specific. 

I think the desired behaviour is for mainstream tools (like https://editor.swagger.io) to accept all output as valid.